src/osd/SnapMapper.cc

```diff
@@ -99,29 +103,30 @@ int OSDriver::get_next(
 
 string SnapMapper::get_prefix(int64_t pool, snapid_t snap)
 {
-  char buf[100];
-  int len = snprintf(
-    buf, sizeof(buf),
-    "%lld_%.*X_",
-    (long long)pool,
-    (int)(sizeof(snap)*2), static_cast<unsigned>(snap));
-  return MAPPING_PREFIX + string(buf, len);
+  return fmt::sprintf("%s%lld_%.*X_",
+		      MAPPING_PREFIX,
+		      pool,
+		      sizeof(snap) * 2,
+		      snap);
```

Comment:
This isn't related to changing the api, it needs its own commit.  It modifies code responsible for generating an ondisk format and should really be revertable on its own.